static/js/components/RecentSources.jsx

```diff
@@ -137,7 +137,31 @@ const RecentSourcesList = ({ sources, styles }) => {
             }
           }
 
-          const imgClasses = source.is_grayscale
+          // Use first thumbnail available
+          const fetchThumbnail = async (index) => {
+            const response = await fetch(source.thumbnails[index]?.public_url, {
+              mode: "no-cors",
+            });
+            return response.status;
+          };
+          let idx = 0;
+          let statusCode = 0;
+          while (idx < source.thumbnails.length - 1) {
```

Comment:
In general, while loops are dangerous and overly verbose and should be replaced by for-loops.

In JavaScript, this loop can be replaced by [`.each`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every), for example.  Alternatively, you can use a for loop and break out of it.

Reply:
I don't think it's possible to break out of a [`forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) loop (I assume that's what you meant). I can replace it with a for loop if you'd like.

Reply:
I meant a `.each` loop---link in comment above.

Reply:
That links to `.every`

Reply:
Which doesn't really apply to this use case

Reply:
You can use `.some` or `.every`.

Reply:
```
source.thumbnails.some((thumb) =>
  ...
  return statusCode == 200;
);
```